sdks/go/pkg/beam/core/runtime/exec/datasource.go

```diff
@@ -63,6 +63,11 @@ type DataSource struct {
 
 	// Whether the downstream transform only iterates a GBK coder once.
 	singleIterate bool
+
+	// state of the SDK with respect to the status of its data channel. If it is true, then the SDK
+	// is waiting for data to be sent to it. If it is false, then the SDK is not ready to take any
```

Comment:
This sentence about the false condition isn't correct. The SDK is basically *always* ready to take data, it just might not be processing it immediately. I'd avoid ascribing additional interpretations to the boolean being false, since all it means is that the SDK isn't blocked on receiving data to make progress.

Reply:
done. thanks!